src/Hit/Git/New.hs

```diff
@@ -0,0 +1,89 @@
+-- | Everything related to the `hit new` command
+
+module Hit.Git.New
+    ( runNew
+    ) where
+
+import qualified Data.Text as T
+import Data.Char (isAlphaNum, isDigit, isSpace)
+
+import GitHub (Issue (issueNumber), IssueNumber (..), unIssueNumber)
+
+import Hit.ColorTerminal (errorMessage, infoMessage, successMessage)
+import Hit.Issue (createIssue, getIssueTitle, mkIssueId)
+import Hit.Git.Common (getUsername)
+
+-- | @hit new@ command.
+runNew :: Bool -> Text -> IO ()
+runNew isIssue issueOrName = do
+    login <- getUsername
+    maybeIssue <- if isIssue then tryCreateNewIssue login else pure Nothing
+    let branchDescription = mkBranchDescription maybeIssue issueOrName
+    title <- displayBranchDescription branchDescription
+    let branchName = login <> "/" <> title
+    "git" ["checkout", "-b", branchName]
+  where
+    tryCreateNewIssue :: Text -> IO (Maybe IssueNumber)
+    tryCreateNewIssue login = do
+        infoMessage $ "Creating issue with title: '" <> issueOrName <> "'"
+        createIssue issueOrName login >>= \case
+            Left err -> do
+                errorMessage "Error creating issue under 'hit new' command!"
+                putTextLn $ show err
+                pure Nothing
+            Right issue -> do
+                let issueNum = issueNumber issue
+                successMessage $ "Successfully created issue number #"
+                    <> show (unIssueNumber issueNum)
+                pure $ Just issueNum
+
+-- QUESTION: should we somehow move this into separate module or split this module
+--           smaller parts?
```

Comment:
No, I think this is okay now :+1:

Reply:
Yep, this was my previous comment from my previous PR. Now it can be deleted 🙂

Reply:
deleted :+1: